src/mono/msbuild/AndroidAppBuilder/ApkBuilder.cs

```diff
@@ -0,0 +1,186 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+public class ApkBuilder
+{
+    public static (string apk, string packageId) BuildApk(
+        string appName,
+        string workplace,
+        string appDir,
+        string entryPointLib,
+        string monoRuntimeHeaders,
+        string sdkPath,
+        string ndkPath,
+        string abi,
+        string minApiLevel,
+        string? buildApiLevel = null,
+        string? buildToolsVer = null)
```

Comment:
This is ugly. Consider changing this method to instance method

Reply:
@marek-safar do you mean to put them to properties?